.github/workflows/main.yml

```diff
@@ -54,9 +56,45 @@ jobs:
         env:
           ENABLE_SEARCH: 'true'
 
+      - name: Initialize Deployment
+        uses: bobheadxi/deployments@v0.6.2
+        id: start-deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: ${{ github.ref_name }}
+
+      - name: Deploy to Vercel
+        id: deploy-to-vercel
+        uses: amondnet/vercel-action@v20
+        with:
+          vercel-project-name: commercetools-docs-kit
+          vercel-token: ${{ secrets.VERCEL_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # vercel-args: '--prod'  # TODO only of on main branch
+          vercel-org-id: ${{ secrets.VERCEL_ORG_ID}}
+          vercel-project-id: ${{ secrets.VERCEL_PROJECT_ID}}
+          alias-domains: | #Optional
+            docs-kit-sha-${{ github.sha }}-commercetools.vercel.app
+            docs-kit-pr-{{PR_NUMBER}}-commercetools.vercel.app
+            docs-kit-branch-{{BRANCH}}-commercetools.vercel.app
+
+      - name: Update Deployment Status
+        uses: bobheadxi/deployments@v0.6.2
+        id: finish_deployment
+        with:
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+          env_url: ${{ steps.deploy-to-vercel.outputs.preview-url }}
+          deployment_id: ${{ steps.start-deployment.outputs.deployment_id }}
+
+      # TODO move this into a separate workflow that holds the static checks only? They can run independently of builds
       - name: Lint text files
         run: yarn run jest --projects jest.{eslint,stylelint,test,text}.config.js --maxWorkers=3 --reporters jest-silent-reporter
 
+      # Tests run after deployment to make sure a preview URL is available for every build
+      # TODO maybe just stop the job here when on the main branch?
```

Comment:
If we decide to deploy first and test after, we can also think of running the tests against the preview URL, instead of localhost. This includes:

* e2e tests
* link checker

Reply:
let's change such things separately.  I prefer local, it's faster and more reliable.